pronto_quadruped/src/ImuBiasLock.cpp

```diff
@@ -48,26 +49,27 @@ ImuBiasLock::ImuBiasLock(const Eigen::Isometry3d& ins_to_body,
   bias_transform_ = Eigen::Isometry3d::Identity();
   gravity_transform_ = Eigen::Isometry3d::Identity();
 
+  gyro_bias_history_.reserve(max_size + 1);
+  accel_bias_history_.reserve(max_size + 1);
 }
 
 RBISUpdateInterface* ImuBiasLock::processMessage(const ImuMeasurement *msg,
                                                  StateEstimator *est)
 {
-
   RBIS prior;
   RBIM prior_cov;
   est->getHeadState(prior, prior_cov);
 
-  current_omega_ = ins_to_body_.rotation()*msg->omega;
-  current_accel_ = ins_to_body_.rotation()*msg->acceleration;
+  current_omega_.noalias() = ins_to_body_.rotation()*msg->omega;
+  current_accel_.noalias() = ins_to_body_.rotation()*msg->acceleration;
   current_accel_corrected_ = current_accel_ - (rotation::skewHat((current_omega_ - previous_omega_) / dt_) + rotation::skewHat(current_omega_)*rotation::skewHat(current_omega_))*ins_to_body_.translation();
   previous_omega_ = current_omega_;
   if(do_record_){
     gyro_bias_history_.push_back(current_omega_);
     accel_bias_history_.push_back(current_accel_corrected_);
     if(gyro_bias_history_.size() > max_size){
-      std::cout << "Stop recording (size too big)" << std::endl;
-      std::cout << gyro_bias_history_.size() << std::endl;
+      std::cout << "Stop recording (size too big)" << "\n";
```

Comment:
Why doing this? Is it really necessary?

Reply:
It's recommended for performance when printing a lot that may slow down a programme, cf. https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rio-endl

Arguably, the gyro bias shouldn't be printed that often due to history size so happy to revert if you prefer

Reply:
well it says "occasionally important". let's just put this in a if debug statement and leave the endl for aesthetics